### ISSUE TITLE
Please add "advanced mode" features.

### ISSUE THREAD

**Issue by Steve132** [bitsai-org/bitsai#1]
I would like to use BitSai as a way to safely and simply access cold storage wallets and paper wallets without installing anything. e.g. on a raspberry pi device.

In order to do this, several options would need to be added.  

First, it's impossible for me to import any of my wallets right now because you *only* support BIP84 and only for a single "account" (account 0).    Please add an "advanced mode" where I can select a wallet format AND full derivation path if I want to (e.g. https://iancoleman.io/bip39/ ).  Its critically important to have at least "wallet path" as a string option and "address type" as a drop down which lets you select from: "Legacy P2PKH","Legacy SegWit (Bip49)","Native Segwit (Bech32/Bip84)<default>" and also the various bip141 options.

Secondly, I'd really like the ability to export and import a transaction from a json string or filebrowser to be signed offline.  

Similarly, I'd need to be able to import either an address or an xpub as a watch-only wallet, and be able to import the corresponding xpriv or private key to be used to sign the transaction. 

I know these are a lot of features, but BitSai is *so* cool and hits a use case.  I'd consider paying a bounty for these features if you need.

**Comment by Steve132** [comment]
Also, I'm a developer if you are willing to add me to the organization or accept pull requests I can start implementing them too.

**Comment by Adoliin** [comment]
Thanks for using BitSai and I am glad it fulfilled that use case :)

> Its critically important to have at least "wallet path" as a string option

Yes, I am planning to add this feature, but this change how the wallet behave and it can no longer detect wallets automatically from just a "seed phrase" so we must implement some type of wallet address fetcher that will try to detect which type of wallet it is.

> "Legacy P2PKH","Legacy SegWit (Bip49)","Native Segwit (Bech32/Bip84)" and also the various bip141 options

This one is pretty complicated, because it will change a lot of the internal APIs .
Currently I am working on unit tests and integration tests so I can't deal with it right now but if you can implement these changes yourself then every pull request is welcomed :)

> Also, I'm a developer if you are willing to add me to the organization or accept pull requests I can start implementing them too

Of course, just add some feature (try to make your git commits as small as possible) and I will test them locally (I am still working on the unit tests). And if they are okay we will add them.

**Comment by Steve132** [comment]
>

Yes, I am planning to add this feature, but this change how the wallet
behave and it can no longer detect wallets automatically from just a "seed
phrase" so we must implement some type of wallet address fetcher that will
try to detect which type of wallet it is.

No, do not implement that.  I'm saying that you should simply be allowed to
*set* it in advanced settings.  As in, you don't have to detect anything
automatically by querying the blockchain i *don't* want that.

I'm saying just when you input a seed phrase have an "advanced settings"
that allows you to manually specify the derivation path and the address
type.  That's it.

**Comment by Steve132** [comment]
> This one is pretty complicated, because it will change a lot of the
internal APIs .

I do not believe so.  The internal APIs should just be using one function
for address validation and decoding, yes? Presumably that function is from
a crypto library (like cryptocoinjs) which supports all bitcoin address
types, otherwise you would not
be able to use the wallet to perform sends.  If it only supports some
address types then that is a much more serious bug.

**Comment by Adoliin** [comment]
> No, do not implement that.  I'm saying that you should simply be allowed to *set* it in advanced settings.  As in, you don't have to detect anything automatically by querying the blockchain i *don't* want that.

Okay, got it :)

> I do not believe so.  The internal APIs should just be using one function for address validation and decoding, yes?
 
By internal APIs I meant BitSai's internal APIs. And yes I am using bitcoinjs as for bitcoin low level instructions.

Anyways, don't hesitate to submit PR for the these features ;)
